cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/scheduler/SlowStartDelayStrategy.java

```diff
@@ -0,0 +1,65 @@
+package com.sequenceiq.cloudbreak.cloud.aws.scheduler;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.waiters.PollingStrategy;
+import com.amazonaws.waiters.PollingStrategyContext;
+
+/**
+ * A polling strategy which is less aggressive early, and gets more aggressive once the
+ * provided expectedRuntime is reached.
+ */
+public class SlowStartDelayStrategy implements PollingStrategy.DelayStrategy {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SlowStartDelayStrategy.class);
+
+    private static final Random RANDOM = ThreadLocalRandom.current();
+
+    private static final int MAX_POLLING_INTERVAL_SECONDS = 60;
+
+    private static final int STABLE_POLLING_INTERVAL_SECONDS_MAX = 5;
+
+    private static final int STABLE_POLLING_INTERVAL_SECONDS_MIN = 1;
+
+    private static final double FIRST_POLL_RATIO = 0.75;
+
+    private static final double SECOND_POLL_RATIO = 0.2;
+
+    private static final double THOUSAND = 1000.0;
+
+    private final int expectedRuntimeSeconds;
+
+    public SlowStartDelayStrategy(int expectedRuntimeSeconds) {
+        this.expectedRuntimeSeconds = expectedRuntimeSeconds;
+    }
+
+    @Override
+    public void delayBeforeNextRetry(PollingStrategyContext pollingStrategyContext) throws InterruptedException {
+        int requestNumber = pollingStrategyContext.getRetriesAttempted();
+
+        // Start polling at 75% of minExpectedRuntime.
+        // Next at 20% of minExpectedRuntime
+        // After this random between min and max
+
+        Double msToWait;
+        if (requestNumber == 0) {
+            msToWait = THOUSAND * Math.min(
+                    Math.max(FIRST_POLL_RATIO * expectedRuntimeSeconds, STABLE_POLLING_INTERVAL_SECONDS_MIN),
+                    MAX_POLLING_INTERVAL_SECONDS);
+        } else if (requestNumber == 1) {
+            msToWait = THOUSAND * Math.min(
+                    Math.max(SECOND_POLL_RATIO * expectedRuntimeSeconds, STABLE_POLLING_INTERVAL_SECONDS_MIN),
+                    MAX_POLLING_INTERVAL_SECONDS);
+        } else {
+            msToWait = THOUSAND * Math.max(STABLE_POLLING_INTERVAL_SECONDS_MIN,
+                    RANDOM.nextInt(STABLE_POLLING_INTERVAL_SECONDS_MAX));
```

Comment:
https://docs.aws.amazon.com/AWSEC2/latest/APIReference/throttling.html is what I could find on AWS EC2 throttling (assuming the same applies for ASGs as well).
I'm not sure how many concurrent operations a single instance of CB is supposed to be able to handle (this would be new AWS cluster launches / scale operations).
How about setting this to a constant 5 seconds (or 5 seconds + jitter in the 1s range... so 4-6 seconds). Can even remove this class and rely on the default FixedDelayStrategy which is part of the AWS SDK.
The current Exponential backoff seems wrong though - since it gets less aggressive towards the time the ASG will be ready.

Reply:
Also, as part of this one thing I wanted to check but did not get a chance to - does the AWS SDK have retries built in. There's several operations outside of polling which would fail if the AWS service were down intermittently.

Reply:
the main issue here is that if a lot of people share the same account it can easily trigger throttling error, if I remember well this is the reason why the backoff strategy was introduced. maybe @sodre90 remembers better

Reply:
Yes, this isn't an issue with a single cluster. But on mow-dev ~100 people use 3 AWS accounts where it's a constant issue for us.